UI/CMakeLists.txt

```diff
@@ -517,6 +517,13 @@ source_group(
 unset(_SOURCES)
 unset(_UI)
 
+get_property(OBS_MODULE_LIST GLOBAL PROPERTY OBS_MODULE_LIST)
+foreach(module ${OBS_MODULE_LIST})
```

Comment:
Use `list(JOIN ..)` instead (https://cmake.org/cmake/help/latest/command/list.html?highlight=list#join) - I guess `\n\t` as glue should work.

Reply:
Would that quote each element though?

Alternatively I suppose it would be possible to make it one comma separated string and split it at runtime rather than relying on CMake to produce a full list.

Reply:
In that case you'd output `"<glued string>"` and make `"\n\t"` the glue.

If we pass one long string, we wouldn't need a separate header and you could pass it as a compiler definition, though I'm afraid we could run afoul of Windows command line length limits then?

Reply:
The limit is 8000-something and a list of all modules is "only" around 350 so we probably still have a decent bit of a breathing room there. I'll try it at least.

Reply:
I've now reworked it to use a define. Seems to work fine on my Windows system.